core/src/avm2/globals/flash/geom/transform.rs

```diff
@@ -173,6 +199,41 @@ pub fn color_transform_to_object<'gc>(
     Ok(object.into())
 }
 
+pub fn matrix3d_to_object<'gc>(
+    matrix: Matrix3D,
+    activation: &mut Activation<'_, 'gc>,
+) -> Result<Value<'gc>, Error<'gc>> {
+    let args = matrix.raw_data.map(Into::into);
+    let object = activation
+        .avm2()
+        .classes()
+        .matrix3d
+        .construct(activation, &args)?;
+    Ok(object.into())
+}
+
+pub fn object_to_matrix3d<'gc>(
+    object: Object<'gc>,
+    activation: &mut Activation<'_, 'gc>,
+) -> Result<Matrix3D, Error<'gc>> {
+    let raw_data = object
+        .get_public_property("rawData", activation)?
+        .coerce_to_object(activation)?;
```

Comment:
This should use `.as_object().expect("rawData cannot be null");` instead of `coerce_to_object`

Reply:
Hi thank you. I basically followed the way of `object_to_matrix` and it seems to use `coerce_*` functions.

But actually, in the line 223-224, I looked for any `() -> Result<_,_>` function but I couldn't find a good one. Do you know any good way?

Reply:
`coerce_to_object` is the one function we're trying to phase out, `as_object` should always be used instead (returns either `Some(object)` for an object or `None` for null, except `rawData` can never be null). What do you mean in your second point? ` -> Result<Matrix3D, Error<'gc>>` looks fine to me.

Reply:
> coerce_to_object is the one function we're trying to phase out

Oh, I see, that makes sense. Fixed: [01c9101](https://github.com/ruffle-rs/ruffle/pull/18810/commits/01c9101479e676070dfba67ec387cc59403d4019)

> What do you mean in your second point?

Ah, sorry. I thought I needed to avoid `.expect()`, but it's probably not the case. Please forget it.